### PR TITLE
Update pipelineTasks in Release-Pipeline to use Git Resolver

### LINKED ISSUE #6379: switch `release-pipeline` to use remote resolution
We updated the `release-pipeline` to take advantage of git resolver for one of the `pipelineTasks` - `publish-images`. This is great and helps the release manager to rely on the task definition from GitHub repo.

Can we update the `release-pipeline` such that rest of the `pipelineTasks` use the same technology?

* [git-clone](https://github.com/tektoncd/pipeline/blob/main/tekton/release-pipeline.yaml#L55)
* [precheck](https://github.com/tektoncd/pipeline/blob/main/tekton/release-pipeline.yaml#L68)
* [unit-tests](https://github.com/tektoncd/pipeline/blob/main/tekton/release-pipeline.yaml#L83)
* [golang-build](https://github.com/tektoncd/pipeline/blob/main/tekton/release-pipeline.yaml#L96)
* [publish-to-bucket](https://github.com/tektoncd/pipeline/blob/main/tekton/release-pipeline.yaml#L146)
* [publish-to-bucket-latest](https://github.com/tektoncd/pipeline/blob/main/tekton/release-pipeline.yaml#L167)

### PR DESCRIPTION
**Pull request by jsminem**
Update `pipelineTasks` in `release-pipeline.yaml` file
Fixes #6379  
/kind misc

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [X] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [X] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [X] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: tekton/release-pipeline.yaml [hunk 1]
```diff
@@ -54,6 +54,16 @@ spec:
...
+        resolver: git
+        params:
+        - name: repo
+          value: pipeline
+        - name: org
+          value: tektoncd
+        - name: revision
+          value: $(params.gitRevision)
+        - name: pathInRepo
+          value: tekton/publish.yaml
```
Comment (afrittoli) [inline tekton/release-pipeline.yaml:66]: The git-clone task is defined in the catalog, not in the pipeline repo.
We could either pull it from there with the git resolver and pin it to a revision that matches the current sha of the tip of the `main` branch. Alternatively we could use the bundle remote resolver and consume it, as we do for in the plumbing repo, from the bundles published to grc.io - see https://github.com/tektoncd/plumbing/blob/2b2d5d06c8ffd885945feae76318ac8a153b3c8d/tekton/ci/jobs/tekton-golang-tests.yaml#L37-L44 for an example

File: tekton/release-pipeline.yaml [hunk 2]
```diff
@@ -66,7 +76,16 @@ spec:
     - name: precheck
       runAfter: [git-clone]
       taskRef:
-        name: prerelease-checks
```
Comment (afrittoli) [inline tekton/release-pipeline.yaml:69]: The name here should not be deleted

File: tekton/release-pipeline.yaml [hunk 3]
```diff
@@ -66,7 +76,16 @@ spec:
...
+        resolver: git
+        params:
+        - name: repo
+          value: pipeline
```
Comment (afrittoli) [inline tekton/release-pipeline.yaml:82]: The `prerelease-check` comes from the plumbing repo. It is deployed together with the pipeline to the cluster when the pipeline definition is installed, so maybe we don't need to change this for now. 
If you wanted to use remote resolution here, this is the URL https://github.com/tektoncd/plumbing/blob/main/tekton/resources/release/base/prerelease_checks.yaml

File: tekton/release-pipeline.yaml [hunk 4]
```diff
@@ -80,6 +99,17 @@ spec:
           subpath: git
     - name: unit-tests
       runAfter: [precheck]
+      taskRef:
```
Comment (afrittoli) [inline tekton/release-pipeline.yaml:102]: We only need 1 `taskRef` per task, this one is missing the task name, the one below in L114 has the name but not the other params.

File: tekton/release-pipeline.yaml [hunk 5]
```diff
@@ -80,6 +99,17 @@ spec:
...
+        resolver: git
+        params:
+        - name: repo
+          value: pipeline
```
Comment (afrittoli) [inline tekton/release-pipeline.yaml:106]: Similar to `git-clone`, this task comes from the catalog.

File: tekton/release-pipeline.yaml [hunk 6]
```diff
@@ -95,6 +125,16 @@ spec:
       runAfter: [precheck]
       taskRef:
         name: golang-build
+        resolver: git
+        params:
+        - name: repo
+          value: pipeline
```
Comment (afrittoli) [inline tekton/release-pipeline.yaml:131]: Similar to `git-clone`, this task comes from the catalog.

File: tekton/release-pipeline.yaml [hunk 7]
```diff
@@ -145,6 +185,16 @@ spec:
       runAfter: [publish-images]
       taskRef:
         name: gcs-upload
+        resolver: git
+        params:
+        - name: repo
+          value: pipeline
```
Comment (afrittoli) [inline tekton/release-pipeline.yaml:191]: Similar to `git-clone`, this task comes from the catalog.

File: tekton/release-pipeline.yaml [hunk 8]
```diff
@@ -166,6 +216,16 @@ spec:
           values: ["true"]
       taskRef:
         name: gcs-upload
+        resolver: git
+        params:
+        - name: repo
+          value: pipeline
```
Comment (afrittoli) [inline tekton/release-pipeline.yaml:222]: Similar to `git-clone`, this task comes from the catalog.

**Review by afrittoli — COMMENTED** [review]
Thank you for your PR.
The only task hosted on pipeline side is the publish one. 
Everything else comes from the catalog or the plumbing repo.
We could still switch to remote resolution, but the revision is not the pipeline one.
For catalog ones, I would suggest to use the bundles, since the version of the task is then visible in the tag.

**Comment by jsminem** [comment]
Thanks so much for the feedback! :pray:

File: tekton/release-pipeline.yaml (hunk 1 above)
Comment (jsminem) [reply tekton/release-pipeline.yaml]: Thanks so much Andrea! This was very helpful. 
I've used the first approach to update the release pipeline. Any feedback would be greatly appreciated! 😊🙏

Comment (jsminem) [reply tekton/release-pipeline.yaml]: Should we also include `package` in params? Like in the example for bundles, https://github.com/tektoncd/catalog?

```
params:
  - name: package
    value: github.com/tektoncd/pipeline
```

Comment (jsminem) [reply tekton/release-pipeline.yaml]: Should I also include `package` in params? Like in the example for bundles, https://github.com/tektoncd/catalog?

```
params:
  - name: package
    value: github.com/tektoncd/pipeline
```

**Review by afrittoli — COMMENTED** [review]
Thank you for all the updates @jsminem, the PR looks good now!
According to "tide", you may need to rebase it though
/approve

**Review by jerop — COMMENTED** [review]
/lgtm

**Pull request merged**